### DISCUSSION TITLE
Print a call stack with variables and return values

### DISCUSSION THREAD

**Discussion by max-arnold** [ionelmc/python-hunter#119] [Q&A] [answered]
Use-case: figure out how a specific piece of code (function) was called in a complex codebase.

How I did that in the past: inserted a snippet like this `import traceback; traceback.print_stack()`

What is not sufficient: without knowing the arguments and return values for each call, it is hard to reason about the differences between multiple calls of the target function

Question: how can I achieve something like this using python-hunter? I.e. show `call` and `return`, but only those that lead to the specified function.

**Comment by max-arnold** [comment]
@ionelmc Sorry for nagging, but do you mind looking at this question as well?

I doubt it is possible to achieve, but maybe you have some ideas.

**Reply by ionelmc** [reply, marked as answer]
It is possible in 3 ways:

* Use a source filter (eg: `source_has='myfunc('` or `fullsource_has='myfunc('`). Obviously this will not trigger in scenarios where your function is passed around as a variable.
* Use a Backlog. Docs here: https://python-hunter.readthedocs.io/en/latest/reference.html#hunter.predicates.Backlog
* Implement your own backlog, eg: https://python-hunter.readthedocs.io/en/latest/cookbook.html#silenced-exception-runtime-analysis

**Reply by ionelmc** [reply]
If docs aren't clear please state what is unclear or how I can improve the docs.
